icu4j/main/common_tests/src/test/java/com/ibm/icu/dev/test/format/NumberFormatTest.java

```diff
@@ -7123,60 +7123,49 @@ public TestData(String unitIdentifier, Integer inputValue, UnitWidth width, ULoc
-        MeasureUnit unit = MeasureUnit.forIdentifier("portion-per-1e9");
-        LocalizedNumberFormatter formatter = NumberFormatter.withLocale(ULocale.ENGLISH).unit(unit)
-                .unitWidth(UnitWidth.FULL_NAME);
-        String formatted1 = formatter.format(1).toString();
-        assertEquals("1 portion per 1e9", "1 part per billion", formatted1);
-
-        String formatted2 = formatter.format(2).toString();
-        assertEquals("1000 portion per 1e9", "2 parts per billion", formatted2);
-
-        String formatted3 = formatter.format(1000000).toString();
-        assertEquals("1000000 portion per 1e9", "1,000,000 parts per billion", formatted3);
-
-        String formatted4 = formatter.format(1000000000).toString();
-        assertEquals("1000000000 portion per 1e9", "1,000,000,000 parts per billion", formatted4);
-    }
-
-    @Test
-    public void TestArbitraryPortionFormat() {
-        class TestData {
+        class TestCase {
             String unitIdentifier;
-            Integer inputValue;
+            String locale;
+            double inputValue;
+            String expectedOutput;
 
-            TestData(String unitIdentifier, Integer inputValue) {
+            TestCase(String unitIdentifier, String locale, double inputValue, String expectedOutput) {
                 this.unitIdentifier = unitIdentifier;
+                this.locale = locale;
                 this.inputValue = inputValue;
+                this.expectedOutput = expectedOutput;
             }
         }
 
-        TestData[] testData = {
-                new TestData("portion-per-1e1", 1),
-                new TestData("portion-per-1e2", 1),
-                new TestData("portion-per-1e3", 1),
-                new TestData("portion-per-1e4", 1),
-                new TestData("portion-per-1e5", 1),
-                new TestData("portion-per-1e6", 1),
-                new TestData("portion-per-1e7", 1),
-                new TestData("portion-per-1e8", 1),
+        TestCase[] testCases = {
+                        new TestCase("portion-per-1e9", "en-US", 1, "1 part per billion"),
+                new TestCase("portion-per-1e9", "en-US", 2, "2 parts per billion"),
+                new TestCase("portion-per-1e9", "en-US", 1000000, "1,000,000 parts per billion"),
+                new TestCase("portion-per-1e9", "de-DE", 1000000, "1.000.000 Milliardstel"),
+                new TestCase("portion-per-1e1", "en-US", 1, "UNKNOWN"),
+                new TestCase("portion-per-1e2", "en-US", 1, "UNKNOWN"),
+                new TestCase("portion-per-1e3", "en-US", 1, "UNKNOWN"),
+                new TestCase("portion-per-1e4", "en-US", 1, "UNKNOWN"),
+                new TestCase("portion-per-1e5", "en-US", 1, "UNKNOWN"),
+                new TestCase("portion-per-1e6", "en-US", 1, "UNKNOWN"),
+                new TestCase("portion-per-1e7", "en-US", 1, "UNKNOWN"),
+                new TestCase("portion-per-1e8", "en-US", 1, "UNKNOWN"),
```

Comment:
Same question/comment as in the C++ version of this change: It looks good, but if all of these `UNKNOWN` cases are things that'll be caught by the `logKnownIssue()` in the code below, you might want to put comments here indicating that.

Reply:
yes, all of them are being caught by `logKnownIssue`,

Okay, I will add a comment in a following PR.